### DISCUSSION TITLE
stable ABI

### DISCUSSION THREAD

**Discussion by Makefile-dot-in** [crablang/crab#97] [Ideas]
this would allow distro maintainers to avoid every crab program containing its own version of tokio it would also allow one to install crab libraries normally as system packages, enable actual dynamic plugin support that won't break if you breathe on it wrong and make life easier for people trying to write pure crab oses. but most importantly, it would allow me to win arguments with C(++) users on discord, making using crabs the obvious choice when i want to oxi-DO.

**Comment by darthdeus** [comment]
I'm not sure if this repo is a meme or real, or if this issue is a meme or real, but at the tiny chance that it is real, please make a stable ABI so that we can have nice things, like ... hot reloading and plugins that could actually work?

**Comment by TCROC** [comment]
> I'm not sure if this repo is a meme or real, or if this issue is a meme or real, but at the tiny chance that it is real, please make a stable ABI so that we can have nice things, like ... hot reloading and plugins that could actually work?

It's real 

Edit: 

And agreed! I want those things too!

**Comment by sylv256** [comment]
See: https://slightknack.github.io/rust-abi-wiki/
We must address these issues before proceeding

**Comment by Gordon01** [comment]
Just `extern "C"`
